thoth/prescriptions_refresh/handlers/pypi_downloads.py

```diff
@@ -0,0 +1,212 @@
+    percentile = downloads.index(packages_total_downloads[package_name]) / len(downloads) * 100
+    return downloads, percentile
+
+
+def _downloads_to_popularity(downloads: int) -> str:
+    """Return the popularity of a package according to its number of downloads."""
+    if downloads < _PYPI_POPULARITY_LOW:
+        return "low"
+    if downloads < _PYPI_POPULARITY_MODERATE:
+        return "moderate"
+    if downloads < _PYPI_POPULARITY_HIGH:
+        return "high"
+    return "very high"
+
+
+def pypi_downloads(prescriptions: "Prescriptions") -> None:
+    """Retrieve the number of downloads for PyPI packages."""
+    _LOGGER.info("Querying number of package downloads available in BigQuery")
+
+    client = bigquery.Client()
+    dataset_id = "pypi_downloads"
+    dataset_id_full = f"{client.project}.{dataset_id}"
+    dataset = bigquery.Dataset(dataset_id_full)
+    dataset = client.create_dataset(dataset)
+
+    job_config = bigquery.QueryJobConfig()
+    job_config.destination = f"{dataset_id_full}.destination_table"
+
+    query = f"""
+    SELECT *
+    FROM `bigquery-public-data.pypi.file_downloads`
+    WHERE DATE(timestamp)
+        BETWEEN DATE_SUB(CURRENT_DATE(), INTERVAL {_PYPI_DOWNLOADS_TIME_INTERVAL} DAY)
+        AND CURRENT_DATE()
+    """
+    query_job = client.query(query=query, job_config=job_config)
+
+    rows = query_job.result()
+    packages_downloads_dict = {}
+    for row in rows:
+        packages_downloads_dict[(row.file.project, row.file.version)] = (
+            packages_downloads_dict.get((row.file.project, row.file.version), 0) + 1
+        )
+
+    # plot statistics:
+
+    # concatenated_package_version_dict = {
+    #     package[0] + " " + package[1]: downloads for package, downloads in packages_downloads_dict.items()
+    # }
+
+    # _LOGGER.info("Creating plots of the number of downloads per package and per package version")
+    # _plot_statistics(packages_downloads_dict)
+    # _plot_statistics(concatenated_package_version_dict, includes_versions=True)
+
+    packages_total_downloads = _packages_total_downloads(packages_downloads_dict)
+    for project_name in prescriptions.iter_projects():
+        prescription_name = prescriptions.get_prescription_name("PackagePopularityWrap", project_name)
+
+        downloads_count, popularity_level = _popularity_level(packages_total_downloads, project_name)
+        package_link = f"https://pypi.org/project/{project_name}"
+
+        package_versions_downloads = {
+            (package[0], package[1]): downloads
+            for package, downloads in packages_downloads_dict.items()
+            if package[0] == project_name
+        }
+        popularity_per_version_summary = ""
+        for package_version, downloads in package_versions_downloads:
+            popularity_per_version_summary += f"'{package_version[0]}' version {package_version[1]} has a {_downloads_to_popularity(downloads)} popularity with {downloads} downloads in the last six months. \n"
```

Comment:
`six` will be misleading here if `THOTH_PRESCRIPTIONS_REFRESH_PYPI_DOWNLOADS_TIME_INTERVAL != 180`. Also in `_PACKAGE_DOWNLOADS_PER_VERSION_PRESCRIPTION_CONTENT`.

Reply:
I'm thinking if this line will not get eventually too large. It might be that `_PACKAGE_DOWNLOADS_PER_VERSION_PRESCRIPTION_CONTENT` will capture this information. What about reporting the most used version here?

Reply:
> What about reporting the most used version here?

Right, this would be more concise and useful :+1: